src/longrange_force_types/force_ewald_cabana_neigh.h

```diff
@@ -0,0 +1,105 @@
+/****************************************************************************
+ * Copyright (c) 2018-2019 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+
+//TODO: I don't really understand this first part
+#ifdef MODULES_OPTION_CHECK
+  if( (strcmp(argv[i], "--force-iteration") == 0) ) {
+    if( (strcmp(argv[i+1], "NEIGH_FULL") == 0) )
+      lrforce_iteration_type = FORCE_ITER_NEIGH_FULL;
+    if( (strcmp(argv[i+1], "NEIGH_HALF") == 0) )
+      lrforce_iteration_type = FORCE_ITER_NEIGH_HALF;
+  }
+  if( (strcmp(argv[i], "--neigh-type") == 0) ) {
+    if( (strcmp(argv[i+1], "NEIGH_2D") == 0) )
+      neighbor_type = NEIGH_2D;
+    if( (strcmp(argv[i+1], "NEIGH_CSR") == 0) )
+      neighbor_type = NEIGH_CSR;
+  }
+#endif
+#ifdef FORCE_MODULES_INSTANTIATION
+    else if (input->lrforce_type == FORCE_EWALD) {
+      bool half_neigh = input->lrforce_iteration_type == FORCE_ITER_NEIGH_HALF;
+      if (input->neighbor_type == NEIGH_2D) {
+        if (half_neigh)
+          lrforce = new ForceEwald<t_verletlist_half_2D>(system,half_neigh);
+        else
+          lrforce = new ForceEwald<t_verletlist_full_2D>(system,half_neigh);
```

Comment:
Return error statement (as in NNP PR) instead of allowing unimplemented full neighbor list

Reply:
Done.